scripts/dockerfiles/run.sh

```diff
@@ -85,6 +90,67 @@ done
         /var/vcap/jobs/{{ $job.Name }}/bin/run
     {{ end }}
 {{ else }}
+    # Run all the scripts called post-start, if any.
+    function post-start-paths()
+    {
+	declare -a fnames
+	idx=0
+	for fname in $(find /var/vcap/jobs/*/bin -name post-start) ; do
+	    fnames[$idx]=$fname
+	    idx=$((idx + 1))
+	done
+	echo ${fnames[*]}
+    }
+
+    psp=$(post-start-paths)
+
+    if [ "X$psp" != X ] ; then
+	# We have post-start scripts to run.
+	# We
+	# - Start a transient monit in the background, treat it as
+	#   non-background however
+	# - Wait for all jobs (except `ready`) to be ok.
+	# - Run the post-start scripts.
+	# - Get rid of the transient monit
+	# - Mark job `ready` as ok, and start the final monit,
+	#   replacing ourselves.
+	# The last two operations are also done when we have no
+	# post-start scripts.
+
+	function monit_report_down() {
+	    # 'monit report down' requires monit 5.18+. BOSH uses monit 5.6.
+	    monit status|grep status|grep -v monitoring|grep -v 'Accessible\|Running'|wc -l
+	}
+
+	dumb-init -- monit -vI &
+	sleep 10
+	while [ $(monit_report_down) -gt 1 ] ; do
+	    echo Still waiting for jobs ...
+	    sleep 10
+	done
+
+	for fname in ${psp} ; do
+	    echo bash $fname
+	    bash $fname
+	done
+
+	# Killing the monit demon now, with all jobs running does not
+	# kill the jobs, only monit itself.
+	monit quit
```

Comment:
Why do we need to do this?
I've read all the comments but I can't tell _why_ we need the _transient_ monit.